purchase_propagate_qty/models/purchase.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2014-2016 Numérigraphe SARL
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+from odoo.tools import float_is_zero
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.multi
+    def write(self, values):
+        res = super().write(values)
+        for line in self:
+            if line.state != 'purchase':
+                continue
```

Comment:
if you replace by ``for line in self.filtered(lambda x: x.state != 'purchase'):`` it could be faster and more synthetic, don't you think ?

Reply:
Actually I don't think so: `filtered()` does a loop, calling a function each time (see https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L5243): This means 2 loops instead of 1 and N additional function calls. I have not benchmarked function calls in Python3 but this used to be much slower.

Reply:
Ok ! Thanks. I didn't know !